### PR TITLE
Autocorrect and autocapitalize should not be property

### LINKED ISSUE #5436: Issues with autoCorrect attribute set as property
Hey. As far as I can see, the `autoCorrect` attribute are set as an HTML property.
https://github.com/facebook/react/blob/master/src/renderers/dom/shared/HTMLDOMPropertyConfig.js#L194

This can give issues with Input elements. Check this :

* Chrome 46.0.2464.0: `autocorrect` in document.createElement('input') === false
* Safari 8.0.7: `autocorrect` in document.createElement('input') === false
* Mobile Safari (iOS 8.4 simulator): `autocorrect` in document.createElement('input') === true

There are identical issues with some other browsers as well.

This is not tested on iOS 9.x

### PR DESCRIPTION
**Pull request by freddyrangel**
Fixes #5436

### CODE REVIEW

**Comment by jimfb** [comment]
The property for autocapitalize is not a boolean, right?  Why should autocapitalize be changed?

cc @zpao

**Comment by zpao** [comment]
The autocorrect property is a boolean, the attribute is not. But the boolean values don't match the attribute values so that's why we need to change it.

autocapitalize… looks like that doesn't matter too much. It's safe to use as an attribute and we can be consistent for these mobile safari-isms.

File: src/renderers/dom/shared/HTMLDOMPropertyConfig.js [hunk 1]
```diff
@@ -192,8 +192,8 @@ var HTMLDOMPropertyConfig = {
      */
     // autoCapitalize and autoCorrect are supported in Mobile Safari for
     // keyboard hints.
-    autoCapitalize: null,
-    autoCorrect: null,
+    autoCapitalize: MUST_USE_ATTRIBUTE,
+    autoCorrect: MUST_USE_ATTRIBUTE,
```
Comment (zpao) [inline src/renderers/dom/shared/HTMLDOMPropertyConfig.js]: Note: now that we're not using properties, can you remove the entries in the `DOMPropertyNames` mapping below. We do not need to add them to the `DOMAttributeNames` mapping.

Comment (freddyrangel) [inline src/renderers/dom/shared/HTMLDOMPropertyConfig.js]: Ah makes sense! Ok done :)

**Comment by zpao** [comment]
Thanks!

**Pull request merged**